packages/drivers/routerlicious-driver/src/test/r11sSocketTests.spec.ts

```diff
@@ -0,0 +1,226 @@
+			new DefaultTokenProvider("jwt"),
+		);
+		resolvedUrl = {
+			type: "fluid",
+			id: "id",
+			url: "https://mock.url/tenantId/documentId",
+			endpoints: {
+				ordererUrl: "ordererUrl",
+				storageUrl: "storageUrl",
+				deltaStorageUrl: "deltaStorageUrl",
+				deltaStreamUrl: "deltaStreamUrl",
+			},
+			tokens: {},
+		};
+		documentService = (await routerliciousDocumentServiceFactory.createDocumentService(
+			resolvedUrl,
+		)) as DocumentService;
+	});
+
+	it("connect_document_error with Token Revoked error", async () => {
+		const errorToThrow = {
+			code: 403,
+			message: "TokenRevokedError",
+			retryAfterMs: 10,
+			internalErrorCode: "TokenRevoked",
+			errorType: DriverErrorTypes.authorizationError,
+			canRetry: false,
+		};
+		const errorEventName = "connect_document_error";
+		socket = new ClientSocketMock({
+			connect_document: { eventToEmit: errorEventName, errorToThrow },
+		});
+
+		let errorOccurred = false;
+		try {
+			await mockSocket(socket as unknown as Socket, async () =>
+				documentService.connectToDeltaStream(client),
+			);
+		} catch (error) {
+			errorOccurred = true;
+			const anyDriverError = error as IAnyDriverError;
+			assert(
+				anyDriverError.errorType === DriverErrorTypes.authorizationError,
+				"Error type should be authorizationError",
+			);
+			assert(
+				anyDriverError.scenarioName === "connect_document_error",
+				"Error scenario name should be connect_document_error",
+			);
+			assert(
+				(anyDriverError as any).internalErrorCode === "TokenRevoked",
+				"Error internal code should be TokenRevoked",
+			);
+		}
+		assert(errorOccurred, "Error should have occurred");
+	});
+
+	it("Socket error with Token Revoked error", async () => {
+		const errorToThrow = {
+			code: 403,
+			message: "TokenRevokedError",
+			retryAfterMs: 10,
+			internalErrorCode: "TokenRevoked",
+			errorType: DriverErrorTypes.authorizationError,
+			canRetry: false,
+		};
+		const errorEventName = "connect_document_success";
+		socket = new ClientSocketMock({
+			connect_document: { eventToEmit: errorEventName },
+		});
+
+		const connection = await mockSocket(socket as unknown as Socket, async () =>
+			documentService.connectToDeltaStream(client),
+		);
+		let error: IAnyDriverError | undefined;
+		connection.on("disconnect", (reason: IAnyDriverError) => {
+			error = reason;
+		});
+
+		// Send Token Revoked error
+		socket.sendErrorEvent(errorToThrow);
+
+		assert(
+			error?.errorType === DriverErrorTypes.authorizationError,
+			"Error type should be authorizationError",
+		);
+		assert(error.scenarioName === "error", "Error scenario name should be error");
+		assert(
+			(error as any).internalErrorCode === "TokenRevoked",
+			"Error internal code should be TokenRevoked",
+		);
+		assert(
+			// eslint-disable-next-line @typescript-eslint/dot-notation
+			documentService["refreshTokenOnConnect"],
+			"Token should be set to refresh on connect",
+		);
```

Comment:
Are we able to instead test that reconnect uses a fresh token? As is, it doesn't look like this test actually confirms whether `refreshTokenOnConnect` does anything

Reply:
Removed this argument based on other suggestion.